File: C#/Observer.cs
using System;
using System.Collections.Generic;
using System.Xml.Linq;


public class Observer
{
    private List<Observer> observers = new List<Observer>();
    private string state;

    public string State
    {
        get { return state; }
        set 
        { 
            state = value;
            NotifyObservers();
        }
    }

    public void Attach(Observer observer)
    {
        observers.Add(observer);
    }

    public void Detach(Observer observer)
    {
        observers.Remove(observer);
    }

    public void NotifyObservers()
    {
        foreach (var observer in observers)
        {
            observer.Update();
        }
    }

    public virtual void Update()
    {
        Console.WriteLine($"Observer has received update: {state}");
    }
}

//ConcreteObserver
public class ConcreteObserver : Observer
{
    private string name;

    public ConcreteObserver(string name, Observer subject)
    {
        this.name = name;
        this.subject = subject;
        subject.Attach(this);
    }

    public override void Update()
    {
        Console.WriteLine($"{name} has received update: {subject.State}");
    }
}

//Main
public class Program
{
    public static void Main(string[] args)
    {
        Observer observer = new Observer();

        ConcreteObserver observer1 = new ConcreteObserver("Observer 1", observer);
        ConcreteObserver observer2 = new ConcreteObserver("Observer 2", observer);

        observer.State = "State 1";
        observer.State = "State 2";

        observer.Detach(observer1);

        observer.State = "State 3";
    }
}

File: Python/Collection/Observer.py
# Der Beobachtbare (Subject)
class Subject:
    def __init__(self):
        self._observers = []
        self._state = None

    @property
    def state(self):
        return self._state

    @state.setter
    def state(self, value):
        self._state = value
        self.notify_observers()

    def attach(self, observer):
        self._observers.append(observer)

    def detach(self, observer):
        self._observers.remove(observer)

    def notify_observers(self):
        for observer in self._observers:
            observer.update()

# Beobachter
class Observer:
    def __init__(self, name, subject):
        self._name = name
        self._subject = subject
        subject.attach(self)

    def update(self):
        print(f"{self._name} hat Update erhalten: {self._subject.state}")

# Hauptfunktion
if __name__ == "__main__":
    subject = Subject()

    observer1 = Observer("Beobachter 1", subject)
    observer2 = Observer("Beobachter 2", subject)

    subject.state = "Zustand 1"
    subject.state = "Zustand 2"

    subject.detach(observer1)

    subject.state = "Zustand 3"
